.github/workflows/build_and_test_cpp_sdk.yml

```diff
@@ -5,6 +5,10 @@ on:
     paths:
     - '.github/workflows/build_and_test_cpp_sdk.yml'
     - 'src/sdk/cpp/**'
+  pull_request:
```

Comment:
Check, if your action can be facilitated by using the following notation:
```
on: [push, pull_request]
  paths
  - '.github/workflows/build_and_test_cpp_sdk.yml'
  - 'src/sdk/cpp/**'
```

Reply:
Unfortunately that isn't (valid) YAML.
Personally I just translate it to json in my head
```
  "on": [
    "push",
    "pull_request"
  ]
```

...and you'll see that you can't add a key (`paths`) to two elements in the array.

But I agree, it would be better to put it in one place, but I don't know how (if) github actions supports that.

(this is the original)

```
  "on": {
    "push": {
      "paths": [
        ".github/workflows/build_and_test_cpp_sdk.yml",
        "src/sdk/cpp/**"
      ]
    },
    "pull_request": {
      "paths": [
        ".github/workflows/build_and_test_cpp_sdk.yml",
        "src/sdk/cpp/**"
      ]
    }
  }
```

Reply:
Please see the example here: https://docs.github.com/en/actions/reference/events-that-trigger-workflows

Reply:
Yes.. that's the documentation I used for creating the github action, and if I understand it correctly.

The the `on:` specifier can contain one of the following:
1. event specified as a string, i.e. `on: push`
2. events specified as an array of strings, i.e. `on: [push, pull_request]`
3. events specified as objects (containing configuration) in a dictionary with the event name as key, i.e.
```
on:
  push:
    <some config>:
    ...
  pull_request:
    <some config>:
    ...
```

And I interpret the [following](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#example-using-multiple-events-with-activity-types-or-configuration):
> If you need to specify activity types or configuration for an event, you must configure each event separately.

as: "there is no way to share configuration, for example "path", between multiple events"

Reply:
Sorry, for not reading that part. You are right :)